modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/snapshot/IgniteSnapshotManager.java

```diff
@@ -1987,7 +1984,7 @@ public List<File> snapshotCacheDirectories(String snpName, @Nullable String snpP
      * @return Snapshot metadata instance.
      */
     public SnapshotMetadata readSnapshotMetadata(File snpDir, String consId) throws IgniteCheckedException, IOException {
-        return readSnapshotMetadata(new File(snpDir, snapshotMetaFileName(consId)));
+        return snpChecker.readSnapshotMetadata(snpDir, consId);
```

Comment:
Why both SnapshotManager and SnapshotChecker has methods `readSnapshotMetadata`? Let's implement them in manager. Checker have reference to the manager though `kctx` object.

Reply:
We should keep the ability to read and validate metas, partitions independently of 'heavy' kernel and managers for external tools. Maybe we need a snapshot file manager.

Reply:
Let's implement required logic for external tools in separate ticket. Now it looks like a code duplication.